dbms/src/Storages/DeltaMerge/tests/gtest_sst_files_stream.cpp

```diff
@@ -464,6 +492,60 @@ try
 }
 CATCH
 
+TEST_F(SSTFilesToDTFilesOutputStreamTest, UpperLayerCancel)
+try
+{
+    auto table_lock = storage->lockStructureForShare("foo_query_id");
+    auto [schema_snapshot, unused] = storage->getSchemaSnapshotAndBlockForDecoding(table_lock, false);
+
+    auto mock_stream = makeMockChild(prepareBlocks(50, 100, /*block_size=*/1));
+    auto abort_flag = std::make_shared<std::atomic_bool>(false);
+    auto stream = std::make_shared<DM::SSTFilesToDTFilesOutputStream<DM::MockSSTFilesToDTFilesOutputStreamChildPtr>>(
+        /* log_prefix */ "",
+        mock_stream,
+        storage,
+        schema_snapshot,
+        FileConvertJobType::ApplySnapshot,
+        /* split_after_rows */ 10,
+        /* split_after_size */ 0,
+        0,
+        abort_flag,
+        *db_context);
+
+    auto sp = SyncPointCtl::enableInScope("before_SSTFilesToDTFilesOutputStream::handle_one");
+    stream->writePrefix();
+    auto t = std::thread([&]() {
+        stream->write();
+    });
+    sp.waitAndPause();
+    abort_flag->store(true, std::memory_order_seq_cst);
+    sp.disable();
+    t.join();
+    stream->writeSuffix();
+    auto files = stream->outputFiles();
+    ASSERT_EQ(true, abort_flag->load(std::memory_order_seq_cst));
+    ASSERT_EQ(1, files.size());
```

Comment:
I think here the `files.size()` should be 0?

Reply:
Should be 1. The write loop will run for one iteration, and then blocked by sync point

Reply:
Should be 1, since it will not trigger the abort-cancel logic.

Reply:
why it will not trigger the cancel logic? Isn't this testcase test that logic?

Reply:
The cancel logic is out side the sst file stream. It is in pre handle snapshot logic, together with other cancel logics.

The testcase is for sst file stream rather than pre handle logic. The prehandle logic namely `preHandleSSTsToDTFiles` is complicated, it involves even schema update, is hard to test.

Reply:
OK